src/theory/bags/bag_solver.h

```diff
@@ -43,12 +43,21 @@ class BagSolver
  private:
   /** apply inference rules for MK_BAG operator */
   void checkMkBag(const Node& n);
+  /**
```

Comment:
I think I see the purpose of this method, you are doing both upwards and downwards closure?
Maybe specify this.

It's also important to be clear about what "known elements in" means, maybe give a pointer to the documnetation of SolverState which explains this.

Reply:
Done. 
I am not sure what upwards and downwards mean here. For example suppose we have terms `n=(bag x c), (bag.count x n)
  (bag.count y n)`. Does downwards here mean `x` and upwards mean `y`?
  Also see this new [commit](https://github.com/CVC4/CVC4/pull/5783/commits/41e97e95b3948a6c8e1d25c5fc7c70d979f5022a)

Reply:
Downwards closure means that from (count x (op A B)), you must reason about (count x A) and (count x B).

Upwards closure means that from (count x A) or (count x B), you must reason about (count x (op A B)).